packages/components/src/checkbox/tests/jest/Checkbox.test.tsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable testing-library/no-node-access */
+/* Using closest to get the label is the best way, even react-aria does this. */
+import { Checkbox } from "../../src/Checkbox.tsx";
+import { CheckboxContext } from "../../src/CheckboxContext.ts";
+import { act, screen, waitFor, render } from "@hopper-ui/test-utils";
+import { createRef } from "react";
+import { userEvent } from "@testing-library/user-event";
+
+describe("Checkbox", () => {
+    it("should render with default class", () => {
+        render(<Checkbox>option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox").closest("label");
+        expect(element).toHaveClass("hop-Checkbox");
+    });
+
+    it("should support custom class", () => {
+        render(<Checkbox className="test">option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox").closest("label");
+        expect(element).toHaveClass("hop-Checkbox");
+        expect(element).toHaveClass("test");
+    });
+
+    it("should support custom style", () => {
+        render(<Checkbox marginTop="stack-sm" style={{ marginBottom: "13px" }} >option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox").closest("label");
+        expect(element).toHaveStyle({ marginTop: "var(--hop-space-stack-sm)", marginBottom: "13px" });
+    });
+
+    it("should support DOM props", () => {
+        render(<Checkbox data-foo="bar">option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox").closest("label");
+        expect(element).toHaveAttribute("data-foo", "bar");
+    });
+
+    it("should support slots", () => {
+        render(
+            <CheckboxContext.Provider value={{ slots: { test: { "aria-label": "test" } } }}>
+                <Checkbox slot="test">option 1</Checkbox>
+            </CheckboxContext.Provider>
+        );
+
+        const checkbox = screen.getByRole("checkbox");
+        const element = checkbox.closest("label");
+        
+        expect(element).toHaveAttribute("slot", "test");
+        expect(checkbox).toHaveAttribute("aria-label", "test");
+    });
+
+    it("should support refs", () => {
+        const ref = createRef<HTMLLabelElement>();
+        render(<Checkbox ref={ref}>option 1</Checkbox>);
+
+        expect(ref.current).not.toBeNull();
+        expect(ref.current instanceof HTMLLabelElement).toBeTruthy();
+    });
+
+    it("should be focused on render when autofocus is true", async () => {
```

Comment:
This test tests react-aria-coimponent behavior and could be removed. Same thing for the other below